Feature/refactoring

Wikitelmex opened the pull request:
- add actions interface
- add bookactions that uses actions interface
- add personactions that uses actions interface
- add rentalactions that uses actions interface
- add UI app that shows and execute actions
- refactor the main method
- create MVC like file structure
- move files to their corresponding folder
- update file "requires"
- refactor actions, into actions/views
- create a views folder
- add several views, that only concerns for display data

wikitelmex1 committed:
Refactoring

- add actions interface
- add bookactions that uses actions interface
- add personactions that uses actions interface
- add rentalactions that uses actions interface
- add UI app that show and execute actions
- refactor main method

wikitelmex1 committed:
refactoring files

- create MVC like file structure
- move files to it corresponding folder
- update file "requires"

mikethreels commented on view/bookactions.rb:
```diff
@@ -0,0 +1,24 @@
+    wait
+  end
+
+  def create
+    puts 'Enter the info of the book you want to add:'
+    print 'Title:'
+    title = gets.chomp
+    print 'Author:'
+    author = gets.chomp
+    @books << Book.new(title, author)
+  end
```
- [x] great job on cleaning up the main.rb file 👍 however the `create` method is not in line with the SOLID principle in particular the single responsibility rule. please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) provided by Microverse for more information.

mikethreels commented on view/personactions.rb:
```diff
@@ -0,0 +1,38 @@
+    wait
+  end
+
+  def create
+    puts 'Enter the info of the person you want to add:'
+    print 'Name:'
+    name = gets.chomp
+    print 'Age:'
+    age = gets.chomp.to_i
+
+    puts 'Is this person a student(S) or a teacher(t)?'
+    case gets.chomp.downcase
+    when 's', ''
+      puts 'Does this student have parent permission (y/N)?'
+      person = Student.new(age, name, parent_permission: gets.chomp.downcase == 'y')
+    when 't'
+      puts 'Which specialization is this teacher?'
+      person = Teacher.new(age, gets.chomp, name)
+    end
+
+    @people << person
+    puts @people.include?(person) ? 'Person added!' : 'Person not added!'
+  end
```
- [x] great job on cleaning up the main.rb file 👍 however the `create` method is not in line with the SOLID principle in particular the single responsibility rule. please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) provided by Microverse for more information.

mikethreels commented on view/rentalactions.rb:
```diff
@@ -0,0 +1,40 @@
+    wait
+  end
+
+  def create
+    print 'Enter the ID of the person you want to add a rental to:'
+    id = gets.chomp.to_i
+    person = @people.find { |p| p.id == id }
+    return if person.nil?
+
+    print 'Enter the ID of the book you want to add a rental to:'
+    book_id = gets.chomp.to_i
+    return if @books[book_id].nil?
+
+    Rental.new(Time.now, @books[book_id], person)
+  end
```
- [x] great job on cleaning up the main.rb file 👍 however the `create` method is not in line with the SOLID principle in particular the single responsibility rule. please have a look at [this article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/solid.md#best-practices-and-solid) provided by Microverse for more information.

mikethreels reviewed (changes requested):
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

wikitelmex1 committed:
summary

- refactor actions, into actions/views
- create views folder
- add several views, that only concerns for display data

D4nielJ reviewed (approved):
## Hello @Wikitelmex, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

Your work left me astonished! Great knowledge of the MVC model, congrats!

![](https://c.tenor.com/tYNRpHAQGmsAAAAC/cheering-hurray.gif)

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can always find me on slack!

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
